### PR TITLE
Closes #305 | Implement dataloader for UIT_ViOCD

### LINKED ISSUE #305: Create dataset loader for UIT-ViOCD
Dataloader name: `uit_viocd/uit_viocd.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?uit_viocd

|     Dataset| uit_viocd |
|-------------|---|
| Description | The UIT-ViOCD dataset includes 5,485 reviews e-commerce sites across four categories: fashion, cosmetics, applications, and phones. Each review is annotated by humans, assigning a label of 1 for complaints and 0 for non-complaints. The dataset is divided into training, validation, and test sets, distributed approximately in an 80:10:10 ratio. |
| Subsets | - |
| Languages | vie |
| Tasks | Text Classification |
| License | Unknown (unknown) |
| Homepage | https://huggingface.co/datasets/tarudesu/ViOCD |
| HF URL | https://huggingface.co/datasets/tarudesu/ViOCD |
| Paper URL | https://ebooks.iospress.nl/DOI/10.3233/FAIA210058 |

### PR DESCRIPTION
**Pull request by Alex-HaochenLi**
Closes #305. 

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

**Review by jen-santoso — CHANGES_REQUESTED** [review]
Hi @Alex-HaochenLi, thank you for implementing the dataloader.

Unfortunately, the dataloader couldn't run here, as `_generate_examples` is missing `split: str` as parameter. 
There are two choices:
1. change line 130 to 
`    def _generate_examples(self, filepath: Path, split: str) -> Tuple[int, Dict]:`
and change `dev` to `val` in line 125
2. remove `split` parameters in line 111, 118, 125.
I would recommend option 2 as parameter `split` is not used in `_generate_examples`

After fixing this, I will continue the review.

**Comment by Alex-HaochenLi** [comment]
Hi @jen-santoso, I have revised the code based on option 2. Please have a check.

**Review by jen-santoso — APPROVED** [review]

**Comment by jen-santoso** [comment]
Hi @Alex-HaochenLi, thank you for the revision. LGTM and let's wait for @danjohnvelasco to review!

**Comment by holylovenia** [comment]
Replacing @danjohnvelasco with @holylovenia.

File: seacrowd/sea_datasets/uit_viocd/uit_viocd.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+_LOCAL = False
+
+_URLS = {
+    "train": "https://huggingface.co/datasets/tarudesu/ViOCD/resolve/main/train.csv?download=true",
+    "val": "https://huggingface.co/datasets/tarudesu/ViOCD/resolve/main/val.csv?download=true",
+    "test": "https://huggingface.co/datasets/tarudesu/ViOCD/resolve/main/test.csv?download=true",
+}
+
+_SUPPORTED_TASKS = [Tasks.INTENT_CLASSIFICATION]
```
Comment (holylovenia) [inline seacrowd/sea_datasets/uit_viocd/uit_viocd.py:44]: Could you please pull from the `main` branch and change this to `Tasks.COMPLAINT_DETECTION`?

**Review by holylovenia — CHANGES_REQUESTED** [review]
Hi @Alex-HaochenLi, everything works well. There's only one minor change requested:

**Comment by Alex-HaochenLi** [comment]
Hi @holylovenia, I have updated the supported tasks. Please have a check.

**Review by holylovenia — APPROVED** [review]
LGTM! Thanks a lot, @Alex-HaochenLi. 🙏
